lib/internal/perf/observe.js

```diff
@@ -476,18 +457,12 @@ function hasObserver(type) {
   return observerCounts[observerType] > 0;
 }
 
-function createBuffer() {
-  return {
-    head: null,
-    tail: null,
-    count: 0,
-  };
-}
-
-function resetBuffer(buffer) {
-  buffer.head = null;
-  buffer.tail = null;
-  buffer.count = 0;
+function resetBuffer(type) {
```

Comment:
this isn't actually more readable than just doing `markEntryBuffer = []` inline?